azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -230,6 +240,11 @@ public void run() throws Exception {
             String.format("Not permitted to proxy as '%s' through Azkaban", effectiveUser)
         );
       }
+      // Set parent directory permissions to <uid>:azkaban so user can write in their execution directory
+      // if the directory is not permissioned correctly already (should happen once per execution)
+      if (!canWriteInCurrentWorkingDirectory(effectiveUser)) {
```

Comment:
It would be good to document this - do you think it'd be best to do so here? Or in a comment?